### PR TITLE
Correct Balanced GC Eden size reporting for Mxbean

### PR DESCRIPTION
**Pull request by LinHu2016**
After some GCs, the number of Free regions could be smaller than desired number of Eden regions, which causes the number of reserved Regions is smaller than 0(reserved Region size = Free Region size - Eden Region size), correct Eden region size = Free region size for the case.

fix: https://github.com/eclipse-openj9/openj9/issues/16413

### CODE REVIEW

**Comment by pshipton** [comment]
@dmitripivkine @amicic

**Comment by amicic** [comment]
Ideally, I'd like to deal with these exceptional cases in core GC code rather than MXBean code. For example, getCurrentEdenSizeInBytes API should already handle the case the there is more less free memory available then what desired Eden size is. There is already a lot of complexity to calculate desired size, so adding this extra logic there is preferred - all that logic would be more centralized. Better than spill out part of that complexity to MXBean.

That said, we already have a couple of more spots that we exceptionally handle in MXBean (race where we can allocate more that desired, and GC-end not properly reporting age 0 regions). So I'm ok we handle this exceptional case here too. But in a long run this MXBean code should be simpler...

File: runtime/gc_vlhgc/SchedulingDelegate.cpp [hunk 1]
```diff
@@ -1357,17 +1357,17 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
 	intptr_t maxEdenChange = 0;
 	uintptr_t maxEdenRegionCount = _extensions->getHeap()->getHeapRegionManager()->getTableRegionCount();
 	bool edenIsVerySmall = (_edenRegionCount * 64) < maxEdenRegionCount;
+	/*
+	 * The heap is fully expanded. Eden will be stealing free regions from the entire heap, without telling the heap to grow.
+	 * Note: When heap is fully expanded, the eden sizing logic knows how much free memory is available in the heap, and knows to not grow too much
+	 */
+	maxEdenChange = freeRegions - _edenRegionCount;
```
[the remark this answers is not in the document]
Comment (LinHu2016) [reply runtime/gc_vlhgc/SchedulingDelegate.cpp:1364]: freeRegions could be smaller than _edenRegionCount.

File: runtime/gc_vlhgc/SchedulingDelegate.cpp [hunk 2]
```diff
@@ -1357,17 +1357,17 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
 	intptr_t maxEdenChange = 0;
 	uintptr_t maxEdenRegionCount = _extensions->getHeap()->getHeapRegionManager()->getTableRegionCount();
 	bool edenIsVerySmall = (_edenRegionCount * 64) < maxEdenRegionCount;
+	/*
```
Comment (amicic) [inline runtime/gc_vlhgc/SchedulingDelegate.cpp:1360]: '/' could start at the line below

**Comment by amicic** [comment]
jenkins test sanity win,aix jdk11

**Review by amicic — APPROVED** [review]

**Pull request merged**